third_party/move/move-command-line-common/src/testing.rs

```diff
@@ -11,6 +11,14 @@ pub const EXP_EXT: &str = "exp";
 /// Extension for expected output files compiled by v2
 pub const EXP_EXT_V2: &str = "v2_exp";
 
+pub fn get_compiler_exp_extension() -> &'static str {
+    if get_move_compiler_v2_from_env() {
```

Comment:
Any tests which are dependent from env variables must be run in their own CI action and cannot run in parallel with other tests and different env vars. Its perhaps OK we do this for e2e tests (like we do already) but for unit tests it is not a good design. Where we have already different exp files (prover, compiler e2e tests) we do not do this via env vars but via test configs.